src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -603,7 +607,9 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
             if (isPublicRoomResult(result)) {
                 const clientRoom = cli.getRoom(result.publicRoom.room_id);
                 const listener = (ev) => {
-                    viewRoom(result.publicRoom.canonical_alias ?? result.publicRoom.room_id, true, ev.type !== "click");
+                    const { publicRoom } = result;
+                    const alias = getDisplayAliasForAliasSet(publicRoom.canonical_alias, publicRoom.aliases);
```

Comment:
See other comment re use of a *display* customisation when making API requests.

Reply:
Dropped in favor of just doing a simple alias check